design/setup.md

```diff
@@ -0,0 +1,104 @@
+# setup
+
+The setup package is responsible for configuring the DC/OS CLI to interact with DC/OS clusters.
+
+## Goals
+
+For a given DC/OS master URL, the package will:
+
+- Download the DC/OS CA bundle (in case of an HTTPS URL).
+- Initiate a login.
+- Create a cluster config with the ACS token, the TLS configuration, and cluster name.
+- Attach the CLI to the newly created cluster config.
+
+## Implementation
+
+The setup is triggered with the following command:
+
+    dcos cluster setup https://dcos.example.com
+
+The URL refers to a DC/OS master node. The command accepts the following flags which only apply for an
+HTTPS URL:
+
+    --ca-certs=<ca-certs>
+        Specify the path to a CA bundle file in PEM format. The CLI will then use it to verify that
+        certificates returned by the cluster have a trusted chain.
+
+    --insecure
+        Do not verify certificates returned by the cluster. With this option, even though the HTTPS
+        protocol is used, there is no guarantee that the remote end is your actual DC/OS cluster.
+        This flag should only be used during development / testing.
+
+    --no-check
+        INSECURE: When the --ca-certs flag is not passed, the CLI will attempt to download the CA
```

Comment:
The wording (`INSECURE` at the beginning,  the last line) is quite different from the documentation of `--insecure`, can we make the two paragraphs more similar?